build.gradle.kts

```
@@ -12,9 +12,21 @@ configurations.all {
 
 plugins {
     kotlin(Plugins.Kotlin.PLUGIN_JVM) version Versions.KOTLIN
+    id(Plugins.DETEKT_PLUGIN) version Versions.DETEKT
     id(Plugins.BEN_MANES_PLUGIN) version Versions.BEN_MANES
 }
 
+tasks.named("detekt") {
+    dependsOn(":test_runner:detekt")
+    dependsOn(":integration_tests:detekt")
+    dependsOn(":flank-scripts:detekt")
+}
+
+tasks.named("clean"){
+    dependsOn(":test_runner:clean")
+    dependsOn(":integration_tests:clean")
+}
+
```

```suggestion
subprojects {
    afterEvaluate {
        if (tasks.findByName("detekt") != null) {
            tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
                dependsOn(tasks["detekt"])
            }
        }
    }
}
```

That would configure detekt task to run before every kotlin compilation (for all subprojects)

---

I think i will apply it here as you mentioned. It will force us to work with the code immediately if its picked up with detekt.